src/rgw/rgw_sal.cc

```diff
@@ -102,30 +105,31 @@ int RGWRadosBucket::remove_bucket(bool delete_children, optional_yield y)
   list_op.params.list_versions = true;
   list_op.params.allow_unordered = true;
 
-  bool is_truncated = false;
-  do {
-    objs.clear();
-
-    ret = list_op.list_objects(max, &objs, &common_prefixes, &is_truncated, null_yield);
-    if (ret < 0)
-      return ret;
+  if (!check_empty(y)) {
```

Comment:
doesn't `check_empty()` have to duplicate this bucket listing? why send a bucket listing op to check whether to send a bucket listing op?

Reply:
Yeah.  I think I was assuming it was cached, but it does a list op.